x-pack/plugins/security_solution/public/timelines/components/modal/actions/new_timeline_button.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonEmpty, EuiContextMenuItem, EuiContextMenuPanel, EuiPopover } from '@elastic/eui';
+import React, { useMemo, useState, useCallback } from 'react';
+import { useCreateTimeline } from '../../../hooks/use_create_timeline';
+import { TimelineType } from '../../../../../common/api/timeline';
+import * as i18n from './translations';
+
+interface NewTimelineButtonProps {
+  /**
+   * Id of the timeline
+   */
+  timelineId: string;
+}
+
+/**
+ * Button that opens a popover with options to create a new timeline or a new timeline template
+ */
+export const NewTimelineButton = React.memo(({ timelineId }: NewTimelineButtonProps) => {
```

Comment:
This component is composing a popover and a button to open it, the name confused me at first. Would it make sense to rename it to `NewTimelinePopoverButton`?

Reply:
I decided to keep the current name for a couple of reasons:
- it is consistent with the other component names in that folder (which I recently refactored)
- because it is only used in the modal, I feel like having the name super descriptive isn't really necessary and doesn't bring that much value
- if we want to have such descriptive names, then I wonder how we would name a component that shows a button, which once clicked shows a popover and then clicking on an option would show a open modal, or a save modal, or some other modals... It could get really complex really quickly

Reply:
Yes, it is important to synthesize names, however too much synthetization leads to confusion, we should find a balance. I don't think we need to explain the entire component in detail in the name, just make it intuitive enough and differentiate it from other components doing similar things. 

I suggested the change mainly because I think the _NewTimelineButton_ component is the one that opens a new timeline when clicked (the one suggested in the comment below). This one adds a step before opening the new timeline, showing a popover with different options (timeline & template) when clicked. That's why I think the _NewTimelinePopoverButton_ name fits it better.

Nevertheless, naming becomes subjective very quickly, I won't push to change it, the current name is good enough for me 👍 
Just wanted to give my 5 cents.